subt/doc/virtual-howto.md

```diff
@@ -31,20 +31,17 @@ docker pull osrf/subt-virtual-testbed:cloudsim_bridge_latest
 It is necessary to run also simulation and bridge in other two terminals:
 - terminal 1
     ```
-    xhost +local:root
```

Comment:
Is `xhost` no longer needed?

Reply:
I am still using it and after power failure this is the place where I look it up, i.e. I would leave it here, please.

Reply:
`xhost` is no longer needed. The `run.bash` does `xauth` and it works. 

@m3d it seems your setup is broken more and more.

Reply:
For who else is it not working without the `xhost` call? Can we have a show of hands?

Reply:
In the current form, the instructions do not work for me when working on a remote machine. "Couldn't connect to X" + a segmentation fault when starting the simulator.

What helps and needs to be highlighted:
* I need to be logged into X on the machine also locally.
* In the remote session, I need to set DISPLAY=:0. Either `export DISPLAY=:0` before `sim.bash`, or adding it to the list of variables when starting `sim.bash`, such as `DISPLAY=:0 ROBOT=... ./subt/script/sim.bash`.

The same is true even when I run with `HEADLESS=true`.

Reply:
Alternatively, when I install `virtualgl`, [configure it](https://cdn.rawgit.com/VirtualGL/virtualgl/2.5/doc/index.html#hd006) **without** restricting access to `vglusers` and log in remotely with `ssh -X` (`-X` needed even for headless mode), I do not need to be logged in locally and set the `DISPLAY` environment variable.

However, it is probably better to stick to the headless mode instead of trying to tunnel the simulation window through ssh. The simulation window ends up being jerky and not very responsive.

Reply:
I am currently exploring solutions with xvfb + virtualgl to run the simulation without active X server and without the need to setup DISPLAY by hand.

I am also exploring how to launch gazebo gui on the local computer and let it attach to the remote simulation - rendering the gui locally and getting only the positions from the gazebo server over ign transport.

But I am not there yet.

Reply:
virtualgl might be able to work without access to the local X server: 
 - https://github.com/VirtualGL/virtualgl/issues/10
 - https://virtualgl.org/Documentation/HeadlessNV

I have not tried yet, but that would be my preferred way to go (xvfb + virtualgl but without local X 3D xserver thus forgoing the xhost stuff).

Reply:
https://groups.google.com/g/virtualgl-users/c/PMcMSXcfxH8 it is available in virtualgl prerelease

Reply:
https://github.com/VirtualGL/virtualgl/issues/10#issuecomment-680390532 description by the author of virtualgl on how to work with the new EGL backend

Reply:
https://github.com/VirtualGL/virtualgl/issues/113#issuecomment-692922737 see the EGL descriptio
https://groups.google.com/g/virtualgl-users/c/YZI3cNaVCME/m/JCAeaqIACgAJ to use xvfb + virtualgl connected to 3D X server to run gazebo

Reply:
One thing I noticed with virtualgl is that the `/tmp/.docker.xauth` file that we create on the way, if it does not exist yet (both in the bash launcher and in the python launcher) became a singleton/global variable. I.e. it breaks things a bit.

Namely, when I `ssh -X`, run `sim.bash`, stop the simulation, disconnect, connect again and try running simulation again, it segfaults, because it fails to connect to X, because the authentication file was already in place, but became obsolete when I disconnected and re-connected. Something similar will probably happen once we start supporting multiple people running simulations in parallel, in case hardware allows us to do such brave things.

Running X server on the workstation, without logging in is not an issue *for me*. I vaguely remember checking xvfb early in the Tunnel Circuit and concluding "nope". It could be that virtualgl was the missing piece though.

How does DARPA/OSRF run the simulator on AWS? I doubt they have X server running, or do they?

In any case, I think we went far beyond the intent of this PR. The question was if the instructions work. They work for me with `DISPLAY=:0` in front of `sim.bash` and being logged in locally, as was the case before with `xhost`.

Reply:
> The question was if the instructions work. They work for me with `DISPLAY=:0` in front of `sim.bash` and being logged in locally, as was the case before with `xhost`.

Is it better with eaca947?

> Running X server on the workstation, without logging in is not an issue _for me_. I vaguely remember checking xvfb early in the Tunnel Circuit and concluding "nope". It could be that virtualgl was the missing piece though.

Actually most likely, at that time, it was correct conclusion. It has become only now possible to use EGL backend in virtualgl (it is implemented only in prerelease version). Otherwise also virtualgl needed the access to the currently running 3D xserver anyway.

I don't understand the section with `ssh -X`. I don't know what this does with `xauth`. But I am not going to loose much time over it anyway, since I think that xvfb+virtualgl with egl backend is the way to go. At that point the 3D X server does not need to be running, so we don't need access to it, so the point is moot.

Reply:
@jisa What do you say?